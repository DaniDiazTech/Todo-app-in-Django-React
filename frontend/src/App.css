body {
    height: 100vh;
    background: #536976;
    /* fallback for old browsers */
    background: -webkit-linear-gradient(to right, #292E49, #536976);
    /* Chrome 10-25, Safari 5.1-6 */
    background: linear-gradient(to right, #292E49, #536976);
    /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
    font-family: 'Rubik', sans-serif;
}


.btn-grad {
    background-color: #4169E1;
    color: whitesmoke;
    transition: 0.5s;
}

.btn-grad:hover {
    color: white;
    background-color: #002aa8d8;
}


#task-container {
    max-width: 600px;
    margin: 0 auto;
    box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
    background-color: #fff;

    margin-top: 100px;
    margin-bottom: 100px;

    justify-content: space-around;
    align-items: flex-start;

}

#form-wrapper {
    position: -webkit-sticky;
    position: sticky;
    top: 0rem;
    border-bottom: 1px solid #e9e9e9;
    background-color: #fff;
    box-shadow: 0 3px 8px rgba(0, 0, 0, 0.25);
    padding: 40px;
}

.flex-wrapper {
    display: flex;
}

.task-wrapper {
    cursor: pointer;
    border-bottom: 1px solid #e9e9e9;
    color: #686868;
    margin: 5px;
    padding: 20px;
}

.task-wrapper:hover {
    background-color: #F0F0F0;
}