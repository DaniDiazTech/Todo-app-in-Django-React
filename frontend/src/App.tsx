import ToDoProvider, { ToDoContext } from './context/ToDoProvider';
import { TaskItem, Form, Spin } from './components';
import { useContext, useEffect } from 'react';
import { getData } from './helpers/getData/getData';

const App = () => {
  const { taskList, setTaskList } = useContext(ToDoContext);

  useEffect(() => {
    getData()
      .then((data) =>
        setTaskList({
          data,
          loading: false,
        })
      )
      .catch((e) => console.warn(e));
  }, []);

  return (
    <div className="container mt-2">
      <h1 className="my-2 text-center primary">Task app</h1>
      <Form />
      <div className="container d-flex justify-content-center align-items-center">
        {taskList.loading ? (
          <Spin />
        ) : (
          <div className="container">
            {taskList.data.map(({ id, nameTask, description, complete }) => (
              <TaskItem
                key={id}
                id={id}
                nameTask={nameTask}
                description={description}
                complete={complete}
              />
            ))}

            {taskList.data.length < 1 && (
              <div className="container h-100">
                <p className="mt-5 primary text-center fs-2">
                  Hurray! You don't have tasks yet
                </p>
              </div>
            )}
          </div>
        )}
      </div>
    </div>
  );
};

export default App;
