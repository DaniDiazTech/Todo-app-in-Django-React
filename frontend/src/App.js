import React, { useEffect, useState } from 'react'
import './App.css';

const App = () => {

    const [task, setTask] = useState({
        id: null,
        nameTask: '',
        description: '',
        completed: false,
        editing: false,
    })

    const [taskList, setaskList] = useState([])

    const handleInputChange = ({ target }) => {
        const { name, value } = target
        setTask({
            ...task,
            [name]: value
        })
    }

    const getTask = async () => {
        try {
            const url = "http://127.0.0.1:8000/api/task-list/"
            const response = await fetch(url)
            const data = await response.json()
            setaskList(data)
        } catch (error) {
            console.error(error);
        }
    }

    useEffect(() => {
        getTask()
    }, [])

    // Send data
    const handleSubmit = (e) => {
        e.preventDefault()
        const url = "http://127.0.0.1:8000/api/task-create/"
        fetch(url,{
            method:'POST',
            headers:{
                'Content-type': 'application/json',
            },
            body: JSON.stringify(task)
        }).then(
            getTask(),
            setTask({
                id: null,
                nameTask: '',
                description: '',
                completed: false,
                editing: false,
            })
        ).catch(error=>console.error(error))
    }

   


    return (
        <div className="container mt-2">
            <form className="container-fluid" onSubmit={handleSubmit}>
                <div className="mb-3">
                    <input
                        type="text"
                        name="nameTask"
                        className="form-control"
                        placeholder="Task name"
                        value={task.nameTask}
                        onChange={handleInputChange}
                    />
                </div>
                <div className="mb-3">
                    <textarea
                        className="form-control"
                        name="description"
                        placeholder="Task description"
                        rows="3"
                        value={task.description}
                        onChange={handleInputChange}
                    ></textarea>
                </div>
                <button className="btn btn-primary">Submit</button>
            </form>
            {
                taskList.map(({ id, nameTask, description}) => (
                   <div key={id} className="border m-2">
                        <p>title: {nameTask}</p>
                    <p>description: {description}</p>
                    <button className="btn btn-primary mr-3">Editar</button>
                    <button className="btn btn-danger">Eliminar</button>
                   </div>
                ))
            }
        </div>
    )
}

export default App;
