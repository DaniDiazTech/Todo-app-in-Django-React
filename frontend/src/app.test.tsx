import '@testing-library/jest-dom';
import { shallow } from 'enzyme'
import App from './App';


describe('Test on <App />', () => {
  const wrapper = shallow(<App />)
  
  test('Should render component', () => {
      expect(wrapper).toMatchSnapshot()
  });

});
