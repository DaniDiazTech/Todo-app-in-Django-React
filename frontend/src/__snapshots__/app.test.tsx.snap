// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <App /> Should render component 1`] = `
<div
  className="container mt-2"
>
  <h1
    className="my-2 text-center primary"
  >
    Task app
  </h1>
  <Memo(Form) />
  <div
    className="container d-flex justify-content-center align-items-center"
  >
    <Spin />
  </div>
</div>
`;
